airbyte-integrations/connectors/source-cart/source_cart/source.py

```diff
@@ -19,206 +20,41 @@
-    contains the endpoints
-        - GET v1/customers
-        - GET v1/employees
-
-    then you should have three classes:
-    `class CartStream(HttpStream, ABC)` which is the current class
-    `class Customers(CartStream)` contains behavior to pull data for customers using v1/customers
-    `class Employees(CartStream)` contains behavior to pull data for employees using v1/employees
-
-    If some streams implement incremental sync, it is typical to create another class
-    `class IncrementalCartStream((CartStream), ABC)` then have concrete stream implementations extend it. An example
-    is provided below.
-
-    See the reference docs for the full list of configurable options.
-    """
-
-    # TODO: Fill in the url base. Required.
-    url_base = "https://example-api.com/v1/"
-
-    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        """
-        TODO: Override this method to define a pagination strategy. If you will not be using pagination, no action is required - just return None.
-
-        This method should return a Mapping (e.g: dict) containing whatever information required to make paginated requests. This dict is passed
-        to most other methods in this class to help you form headers, request bodies, query params, etc..
-
-        For example, if the API accepts a 'page' parameter to determine which page of the result to return, and a response from the API contains a
-        'page' number, then this method should probably return a dict {'page': response.json()['page'] + 1} to increment the page count by 1.
-        The request_params method should then read the input next_page_token and set the 'page' param to next_page_token['page'].
-
-        :param response: the most recent response from the API
-        :return If there is another page in the result, a mapping (e.g: dict) containing information needed to query the next page in the response.
-                If there are no more pages in the result, return None.
-        """
-        return None
+from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator
 
-    def request_params(
-        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
-    ) -> MutableMapping[str, Any]:
-        """
-        TODO: Override this method to define any query parameters to be set. Remove this method if you don't need to define request params.
-        Usually contains common params e.g. pagination size etc.
-        """
-        return {}
+from .streams import CustomersCart, OrderPayments, Orders, Products
 
-    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        """
-        TODO: Override this method to define how a response is parsed.
-        :return an iterable containing each record in the response
-        """
-        yield {}
 
+class CustomHeaderAuthenticator(HttpAuthenticator):
```

Comment:
You don't need to create custom header authenticator. You can use TokenAuthenticator, it accept auth_header parameter.

Reply:
actually, I cannot do that without overriding get_auth_header method. It accepts prefix `auth_method` (Bearer, Basic etc) then it uses whitespace, then token. Even if set it to empty string I will have leading whitespace and error returned by API `InvalidHeader('Invalid return character or leading space in header: X-AC-Auth-Token`. So I will need to override both methods (init and get_auth_header) and it does not differ from current one, even make it harder to understand